services/src/pro/contexts/postgres.rs

```diff
@@ -58,6 +68,34 @@ where
             user_db: Arc::new(RwLock::new(PostgresUserDb::new(pool.clone()))),
             project_db: Arc::new(RwLock::new(PostgresProjectDb::new(pool.clone()))),
             workflow_registry: Arc::new(RwLock::new(PostgresWorkflowRegistry::new(pool.clone()))),
+            dataset_db: Default::default(),
+            thread_pool: Default::default(),
+        })
+    }
+
+    pub async fn new_with_data(
+        config: Config,
+        tls: Tls,
+        dataset_defs_path: PathBuf,
+        provider_defs_path: PathBuf,
+    ) -> Result<Self> {
+        let pg_mgr = PostgresConnectionManager::new(config, tls);
+
+        let pool = Pool::builder().build(pg_mgr).await?;
+
+        Self::update_schema(pool.get().await?).await?;
+
+        let mut dataset_db = ProHashMapDatasetDb::default();
+        add_datasets_from_directory(&mut dataset_db, dataset_defs_path).await;
+        add_providers_from_directory(&mut dataset_db, provider_defs_path.clone()).await;
+        add_providers_from_directory(&mut dataset_db, provider_defs_path.join("pro")).await;
+
+        Ok(Self {
+            user_db: Arc::new(RwLock::new(PostgresUserDb::new(pool.clone()))),
+            project_db: Arc::new(RwLock::new(PostgresProjectDb::new(pool.clone()))),
+            workflow_registry: Arc::new(RwLock::new(PostgresWorkflowRegistry::new(pool.clone()))),
+            dataset_db: Arc::new(RwLock::new(dataset_db)),
+            thread_pool: Default::default(),
```

Comment:
Could we write a test that creates a context, stores something, drops the context, creates a new context, and retrieves the workflow?
Then, could see if it works.

Reply:
We already test that we can write to and read from the database. I am not sure why we want to test whether the database is able to persistently stored data.

Reply:
Well, we distribute things into different handlers user data, workflows, etc.
This way we see what we can retrieve and what not. For instance, currently, it should fail to upload a dataset, shut down the Postgres context and restore it.

Reply:
f19d3a9f13266c3023348d6628595a668f06b2a2